code/modules/mob/living/carbon/human/human.dm

```diff
@@ -584,6 +584,8 @@
 		return get_id_name("Unknown")
 	if( head && (head.flags_inv&HIDEFACE) )
 		return get_id_name("Unknown")		//Likewise for hats
+	if(mind && mind.vampire && (VAMP_SHADOW in mind.vampire.powers) && mind.vampire.ismenacing)

```

Comment:
this breaks encapsulation but you don't have much of a choice unless you want to refactor this code too :V